test/common/tst_info.c

```diff
@@ -16,11 +16,22 @@
 struct tst_stat {
     atomic_uint fail;
     atomic_uint pass;
+    atomic_uint skip;
 };
 
 /* Global test statistics updated by tst_end(). */
 static struct tst_stat tst_stats;
 
+int
+tst_exit_code(void)
+{
+    if (atomic_load(&tst_stats.fail))
+        return EXIT_FAILURE;
+    else if (atomic_load(&tst_stats.skip))
+        return EXIT_SKIPPED;
+    return EXIT_SUCCESS;
```

Comment:
If running multiple tst_start()/tst_end() in a single test would the above prove wrong?

I do the following in the same test function called from main:
tst = tst_start("foobar test1"); 
tst_end(tst, SUCCESS);

tst = tst_start("foobar test2"); 
tst_end(tst, SKIP);

tst = tst_start("foobar test3"); 
tst_end(tst, SUCCESS);

The above would report test skipped, but two of the tests passed. Because tst_stats is global for all tests. What
I would expect to see the in above is two passed and one skipped being printed.

The metrics test would work as it only has one set of tst_start/tst_end in the single test.

Reply:
I assumed the test results would be printed for the log, but that does not matter in this question.

If a test reports skipped on exit, it is a bit mis-leading to me, but I can accept it.

Reply:
Using TAP (rather than, or as an option instead of, "exitcode") for reporting test statuses would allow trivially emitting different statuses per subtest. So you could indeed preserve this information.